opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -479,6 +482,57 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if d.HasChange("nic") {
+
+		log.Printf("[INFO] Update NIC configuration")
+
+		old, new := d.GetChange("nic")
+		attachedNicsCfg := old.([]interface{})
+		newNicsCfg := new.([]interface{})
+
+		timeout := d.Get("timeout").(int)
+
+		// wait for the VM to be ready for attach operations
+		_, err = waitForVMState(vmc, timeout, vmNICUpdateReadyStates...)
```

Comment:
Why doing this while for the moment the VM is already supposed to be ready ? 
It wait for 10s for nothing

Reply:
This is paranoïa. Same code for disk, I'll remove both state check.

Reply:
Yes.
When you run a new plan and apply it means previous run was OK. In our case it means VM are Running.
If the VM are are not running OpenNebula will refuse your change and then you will face an error: this is the expected behavior.
So you don't need to perform this check before